### ISSUE TITLE
Bounded TypeVar with TypedDict doesn't work correctly

### ISSUE THREAD

**Issue by bogdandm** [python/mypy#8576]
TypedDict does not work correctly as `bound` argument for TypeVar:

```
from typing import TypeVar, Generic
from typing_extensions import TypedDict


class BaseDict(TypedDict, total=False):
	foo: int

_DICT_T = TypeVar('_DICT_T', bound=BaseDict)

class SomeGeneric(Generic[_DICT_T]):
	def __init__(self, data: _DICT_T) -> None:
		self._data: _DICT_T = data

	def set_state(self) -> None:
		self._data['foo'] = 1  # error: Unsupported target for indexed assignment

	def set_something_else(self) -> None:
		raise NotImplementedError

class SomeUserData(BaseDict):
	uid: int

class Implementation(SomeGeneric[SomeUserData]):
	def set_something_else(self) -> None:
		self._data['uid'] = 10  # It is ok
		self._data['undefined'] = 10  # This is not - TypedDict "UserData" has no key 'undefined'
```
I trying to use bound TypeVar as default value for Generic because I need to extend TypedDict with some fields for every subclass and use this fields at overridden methods.

**Comment by ilevkivskyi** [comment]
IIUC the actual issue is the false positive `# error: Unsupported target for indexed assignment`.

I think this should be not hard to fix. Looks like a missing corner case in type checking indexed assignments.
